src/System.Management.Automation/CoreCLR/CorePsPlatform.cs

```diff
@@ -561,20 +561,255 @@ internal static int NonWindowsGetProcessParentPid(int pid)
         // the string perror would give you for the last set value of errno.
```

Comment:
@JamesWTruher, your last commit had 3 failures in `PowerShell-CI-macos`
UnixFileSystem additions.Validation of additional properties on file system objects.Should retrieve the user name for the file
```powershell
You cannot call a method on a null-valued expression.
at <ScriptBlock>, /Users/runner/runners/2.160.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/UnixStat.Tests.ps1: line 76
76:             $user = (/bin/ls -ld /tmp/foo).split(" ",[System.StringSplitOptions]"RemoveEmptyEntries")[2]

```
UnixFileSystem additions.Validation of additional properties on file system objects.Should retrieve the group name for the file
```powershell
You cannot call a method on a null-valued expression.
at <ScriptBlock>, /Users/runner/runners/2.160.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/UnixStat.Tests.ps1: line 82
82:             $group = (/bin/ls -ld /tmp/foo).split(" ",[System.StringSplitOptions]"RemoveEmptyEntries")[3]

```
Get-FormatData.Can get format data requiring '-PowerShellVersion 5.1'
```powershell
Expected a collection with size 4, but got collection with size 5 @(System.Management.Automation.FormatViewDefinition, System.Management.Automation.FormatViewDefinition, System.Management.Automation.FormatViewDefinition, System.Management.Automation.FormatViewDefinition, System.Management.Automation.FormatViewDefinition).
at <ScriptBlock>, /Users/runner/runners/2.160.0/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Get-FormatData.Tests.ps1: line 18
18:         $format.FormatViewDefinition | Should -HaveCount 4
```